tools/crm_resource_runtime.c

```diff
@@ -965,39 +965,24 @@ generate_resource_params(pe_resource_t *rsc, pe_node_t *node,
 
 bool resource_is_running_on(pe_resource_t *rsc, const char *host)
 {
-    bool found = TRUE;
-    GList *hIter = NULL;
-    GList *hosts = NULL;
+    bool retval = false;
+    GList *lst = NULL;
 
-    if(rsc == NULL) {
-        return FALSE;
-    }
-
-    rsc->fns->location(rsc, &hosts, TRUE);
-    for (hIter = hosts; host != NULL && hIter != NULL; hIter = hIter->next) {
-        pe_node_t *node = (pe_node_t *) hIter->data;
-
-        if(strcmp(host, node->details->uname) == 0) {
-            crm_trace("Resource %s is running on %s\n", rsc->id, host);
-            goto done;
-        } else if(strcmp(host, node->details->id) == 0) {
-            crm_trace("Resource %s is running on %s\n", rsc->id, host);
-            goto done;
-        }
+    if (host == NULL && rsc->running_on == NULL) {
+        crm_trace("Resource %s is not running\n", rsc->id);
+        return false;
```

Comment:
I'm not sure this check is worthwhile, the long-named function should handle it (considering the children's running_on too)

Reply:
I can't take it anymore.  I'm renaming that function to pe__rsc_running_on_any.  The fact that it's on a node should be obvious.  That's where resources run.  And we know there's a list because that's one of the arguments.  Plus, "any" should clue you in to that.

Reply:
Fixed.